apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+    // conversion but only after the type of the thread is set
+    // (afterEnterThread)
+    if (!prevPanel || prevPanel.panel !== 'composer') {
+      this.enableConvertNoticeBanners();
+    }
+
+    if (!this.isConversationPanel(args.id, prevPanel)) {
+      this.initializeRendering();
+    }
+
+    // Call button should be shown only for non-email single-participant thread
+    if (Threads.active.participants.length === 1 &&
+        (!Settings.supportEmailRecipient ||
+         !Utils.isEmailAddress(Threads.active.participants[0]))) {
+      this.callNumberButton.classList.remove('hide');
+    }
+
+    return this.updateHeaderData();
+  },
+
+  afterEnter: function conv_afterEnter(args) {
+    var next = args.meta.next.panel;
+    switch (next) {
+      case 'composer':
+        return this.afterEnterComposer(args);
+      case 'thread':
+        return this.afterEnterThread(args);
+      default:
+        console.error(
+          'afterEnter was called with an unexpected panel name:',
+          next
+        );
+        return Promise.reject();
+    }
+  },
+
+  afterEnterComposer: function conv_afterEnterComposer(args) {
+    // TODO Bug 1010223: should move to beforeEnter
+    if (args.activity) {
+      this.handleActivity(args.activity);
+    } else if (args.draftId) {
+      this.handleDraft(+args.draftId);
+    } else {
+      this.recipients.focus();
+    }
+
+    this.emit('visually-loaded');
+
+    // not strictly necessary but better for consistency
+    return Promise.resolve();
+  },
+
+  afterEnterThread: function conv_afterEnterThread(args) {
+    var threadId = +args.id;
+
+    var prevPanel = args.meta.prev;
+
+    if (!this.isConversationPanel(threadId, prevPanel)) {
+      this.renderMessages(threadId);
+
+      // Populate draft if there is one
+      // TODO merge with handleDraft ? Bug 1164431
+      Drafts.request().then(() => {
+        this.draft = Threads.get(threadId).getDraft();
+
+        if (this.draft) {
+          this.draft.isEdited = false;
+
+          Compose.fromDraft(this.draft);
+        }
+      });
+    }
+
+    // Let's mark thread only when inbox is fully rendered and target node
+    // is in the DOM tree.
+    App.whenReady().then(function() {
+      // We use setTimeout (macrotask) here to allow reflow happen as soon as
+      // possible and to not interrupt it with non-critical task since Promise
+      // callback only (microtask) won't help here.
+      setTimeout(
+        () => InboxView.markReadUnread([threadId], /* isRead */ true)
+      );
+    });
+
+    // Enable notifications redirected from composer only after the user enters.
+    if (prevPanel && prevPanel.panel === 'composer') {
+      this.enableConvertNoticeBanners();
+    }
+
+    if (args.focusComposer) {
+      Compose.focus();
+    }
+
+    return Utils.closeNotificationsForThread(threadId);
+  },
+
+  beforeLeave: function conv_beforeLeave(args) {
```

Comment:
```disableConvertNoticeBanners``` is not needed for separated views.

```cancelEdit``` and attachment disposal code are specific to ConversationView only.

```cleanFields``` are for both.

So looks like can be easily separated between two views (if we really need these cleanups for the case when views are separated :) since entire view will be unloaded).